pootle_pytest/factories.py

```diff
@@ -41,3 +45,110 @@ class AgreementFactory(factory.django.DjangoModelFactory):
 
     class Meta(object):
         model = 'staticpages.Agreement'
+
+
+class DirectoryFactory(factory.django.DjangoModelFactory):
+
+    @factory.lazy_attribute
+    def pootle_path(self):
+        if self.parent is None:
+            return "/"
+        return (
+            "%s/%s"
+            % (self.parent.pootle_path.rstrip("/"),
+               self.name))
+
+    class Meta:
+        model = 'pootle_app.Directory'
+        django_get_or_create = ("pootle_path", )
+
+
+class LanguageFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = 'pootle_language.Language'
+        django_get_or_create = ("code", )
+
+    code = factory.Sequence(lambda n: 'language%s' % n)
+    fullname = factory.Sequence(lambda n: 'Language %s' % n)
+
+
+class ProjectFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = 'pootle_project.Project'
+        django_get_or_create = ("code", )
+
+    code = factory.Sequence(lambda n: 'project%s' % n)
+    fullname = factory.Sequence(lambda n: 'Project %s' % n)
+    pootle_path = factory.LazyAttribute(lambda p: "/projects/%s" % p.code)
+
+
+class StoreFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = 'pootle_store.Store'
+        django_get_or_create = ("pootle_path", )
+
+    @factory.lazy_attribute
+    def pootle_path(self):
+        return ("%s/%s"
+                % (self.translation_project.pootle_path.rstrip("/"),
+                   self.name))
+
+    name = factory.Sequence(lambda n: 'store%s.po' % n)
+    parent = factory.LazyAttribute(
+        lambda s: s.translation_project.directory)
```

Comment:
@unho 

```
(Pdb) Unit.objects.all()[0].store.parent.pootle_path
u'/af/terminology/'
```
this is taken from livestaging - so the factory code is correct in how it constructs stores